### PR TITLE
feat: add the docs for CLI interface

### PR DESCRIPTION
**Pull request by abhi18av**
This PR adds the `Command Line Interface` section in the docs.

- CLI top-level options

image

- CLI top-level commands

image

### CODE REVIEW

File: docs/cli.rst [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+| -keep-logs, -k            |   false    | Removes only temporary files but retains execution log entries and metadata.   |                                           
++---------------------------+------------+--------------------------------------------------------------------------------+
+| -quiet, -q                |   false    | Do not print names of files removed.                                           |
++---------------------------+------------+--------------------------------------------------------------------------------+
+
+
+
+Examples
+^^^^^^^^^^^^^^^^^^^^
+
```
Comment (abhi18av) [inline docs/cli.rst:132]: Please let me know if we're open to adding the Asciinema `Shell casts` in `Usage` section. 

These GIFs can improve the beginner friendliness of the docs and this way we can cover the scenario and provide more context for any command.

[asciicast](https://asciinema.org/a/363719)

File: docs/cli.rst [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+Description
+^^^^^^^^^^^^^^^^^^^^
+Print executions log and runtime info
+
+
+Usage
+^^^^^^^^^^^^^^^^^^^^
+
+The ``log`` command can be invoked like so::
```
Comment (abhi18av) [inline docs/cli.rst:568]: Could you please guide me on the usage of this command - I've tried a few permutations and I only seem to generate the same result.

A couple of examples here would be great 🆘

**Review by abhi18av — COMMENTED** [review]
Highlighted areas where I'd appreciate some guidance.

File: docs/cli.rst (hunk 2 above)
Comment (evanfloden) [reply docs/cli.rst:572]: See 11.4. Execution provenance at https://seqera.io/training/.

File: docs/cli.rst [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+
+
+    $ nextflow
+    Usage: nextflow [options] COMMAND [arg...]
+
+    Options:
+    -C
+        Use the specified configuration file(s) overriding any defaults
+    -D
+        Set JVM properties
```
[the remark this answers is not in the document]
Comment (evanfloden) [reply docs/cli.rst]: We should be clear that these are options for the Nextflow base command and these options are applied when using the child commands (e.g. `nextflow -C nextflow.config run rnaseq.nf`)

File: docs/cli.rst (hunk 1 above)
Comment (evanfloden) [reply docs/cli.rst:132]: Let's leave this out for now.

**Comment by evanfloden** [comment]
For usage, we can remove the line.

`The XXX command can be invoked like so::`

File: docs/cli.rst (hunk 1 above)
Comment (abhi18av) [reply docs/cli.rst:132]: Okey dokey 👍

File: docs/cli.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (abhi18av) [reply docs/cli.rst]: Agreed - let me think about to how put it forward nicely.

File: docs/cli.rst (hunk 2 above)
Comment (abhi18av) [reply docs/cli.rst:572]: Gotcha!

File: docs/cli.rst [hunk 4]
```diff
@@ -0,0 +1,988 @@
...
+
+
+Examples
+^^^^^^^^^^^^^^^^^^^^
+
+
+--------------------
+kuberun       
+--------------------
+
```
Comment (abhi18av) [inline docs/cli.rst:395]: @evanfloden I'd like to confirm, for this iteration, how extensive should be the `EXAMPLES` section for `kuberun` and `run`.

Here are the possible choices:


- Should we document everything extensively right now as the `CLI` doc would probably be referenced quite frequently?

- Should we focus only on the most frequent use-cases and add more examples as per user requests?

File: docs/cli.rst [hunk 5]
```diff
@@ -0,0 +1,988 @@
...
+| -help, -h                 |  false     | Print the command usage.                                                       |
++---------------------------+------------+--------------------------------------------------------------------------------+
+| -o                        |  text      | Output format, either ``text``, ``json`` or ``yaml``.                          |
++---------------------------+------------+--------------------------------------------------------------------------------+
+
+
+
+Examples
+^^^^^^^^^^^^^^^^^^^^
+
```
Comment (abhi18av) [inline docs/cli.rst:390]: While trying out the options for the `info` command, I couldn't make use of the `-o` output feature.
```
(base) Abhinavs-MacBook-Pro:_resources eklavya$ nextflow info -o yaml
  Version: 20.07.1 build 5412
  Created: 24-07-2020 15:18 UTC (20:48 IDT)
  System: Mac OS X 10.15.6
  Runtime: Groovy 2.5.11 on OpenJDK 64-Bit Server VM 1.8.0_192-b01
  Encoding: UTF-8 (UTF-8)

(base) Abhinavs-MacBook-Pro:_resources eklavya$ nextflow info
  Version: 20.07.1 build 5412
  Created: 24-07-2020 15:18 UTC (20:48 IDT)
  System: Mac OS X 10.15.6
  Runtime: Groovy 2.5.11 on OpenJDK 64-Bit Server VM 1.8.0_192-b01
  Encoding: UTF-8 (UTF-8)
```

I've also tried coupling it with `-d` detailed option but the result always seems to be printed out in the `text` format. 

If this is the expected behavior, then perhaps I'm missing some info here 🤔

**Review by abhi18av — COMMENTED** [review]
Highlighted `info` and `kuberun` related aspects which need some clarifications.

File: docs/cli.rst (hunk 4 above)
Comment (abhi18av) [reply docs/cli.rst:952]: I've opted for the most frequent ones and would add more in case there is user request or I get more ideas.

**Review by evanfloden — APPROVED** [review]

**Comment by abhi18av** [comment]
I'm closing this PR since there's another PR which builds upon this work.

https://github.com/nextflow-io/nextflow/pull/1759

**Pull request closed without merging**